tss463_van.cpp

```diff
@@ -215,23 +220,27 @@ void TSS463_VAN::registers_set(uint8_t address, const uint8_t values[], uint8_t
 
     TSS463_SELECT();
 
-    _delay_ms(4);
+    delayMicroseconds(1);//at 8MHZ max speed (4 clocks XTAL)
+    //_delay_ms(4);
     //At the beginning of a transmission over the serial interface, the first byte is the address of the TSS463C register to be accessed
     res = spi_transfer(address);
     if ( res != ADDR_ANSW)
-    error++;
-    _delay_ms(8);
+		error++;
+    delayMicroseconds(2);//at 8MHZ max speed (8 clocks XTAL)
+    //_delay_ms(8);
     //The next byte transmitted is the control byte that determines the direction of the communication
     res = spi_transfer(WRITE);
     if (res != CMD_ANSW)
     error++;
-    _delay_ms(15);
+    delayMicroseconds(4);//at 8MHZ max speed (15 clocks XTAL)
+    //_delay_ms(15);
   
     for (i = 0; i < count; i++)
     {
         spi_transfer(values[i]);
+		delayMicroseconds(3);//at 8MHZ max speed (12 clocks XTAL)
```

Comment:
Based on the datasheet it isn't needed to wait after every byte in the for loop. It would be enough to wait after the transmit finished (outside the loop).